### ISSUE TITLE
[Dynamo] Graph break on str arg to torch function

### ISSUE THREAD

**Issue by eellison** [pytorch/pytorch#107970]
### 🐛 Describe the bug

Add this diff:
```
diff --git a/test/inductor/test_torchinductor_opinfo.py b/test/inductor/test_torchinductor_opinfo.py
index e886e52ce1e..fb9784e489b 100644
--- a/test/inductor/test_torchinductor_opinfo.py
+++ b/test/inductor/test_torchinductor_opinfo.py
@@ -206,7 +206,7 @@ inductor_expected_failures_single_sample["cpu"] = {
     "complex": {f16},
     "exponential": {f16},
     "geometric": {f16},
-    "linalg.eigh": {f32, f64},
+    # "linalg.eigh": {f32, f64},
     "linalg.eigvalsh": {f32, f64},
     "log_normal": {f16},
     "masked_scatter": {f16, f32, f64},
@@ -459,7 +459,7 @@ class TestInductorOpInfo(TestCase):
     @torch._inductor.config.patch(
         {"implicit_fallbacks": False, "triton.autotune_pointwise": False}
     )
-    @collection_decorator
+    # @collection_decorator
     def test_comprehensive(self, device, dtype, op):
         torch._dynamo.reset()
         with torch.no_grad():
@@ -589,6 +589,7 @@ class TestInductorOpInfo(TestCase):
                     )
 
         except Exception as e:
+            raise e
             known_failure = False
             if dtype in inductor_should_fail_with_exception[device_type].get(
                 op_name, set()
```

Run `python /home/eellison/pytorch/test/inductor/test_torchinductor_opinfo.py -k  eigh_cpu_float32`

>   File "/home/eellison/pytorch/torch/_dynamo/variables/torch.py", line 732, in call_function
    *proxy_args_kwargs(args, kwargs),
  File "/home/eellison/pytorch/torch/_dynamo/utils.py", line 497, in proxy_args_kwargs
    raise unimplemented(
  File "/home/eellison/pytorch/torch/_dynamo/exc.py", line 172, in unimplemented
    raise Unsupported(msg)
torch._dynamo.exc.Unsupported: call_function args: TensorVariable() UserDefinedObjectVariable(str_)

This should be a simple fix. it seems like it's failing on converting the str type to value for running the op:

`torch.linalg.eigh(A, UPLO="U")`

### Versions

master

related: https://github.com/pytorch/pytorch/issues/105534

cc @mruberry @rgommers @voznesenskym @penguinwu @anijain2305 @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @Xia-Weiwen @wenzhe-nrv @jiayisunx @chenyang78 @aakhundov

**Comment by anijain2305** [comment]
For anyone looking at this, we should investigate why str is getting converted into a `UserDefinedObjectVariable` and not a `ConstantVariable`.

**Comment by Fidget-Spinner** [comment]
I'm looking into this!

**Comment by Fidget-Spinner** [comment]
The bug seems to be that `numpy.random.choice` has a different behavior under `torch.compile` than under eager. Specifically, it does not support an array of strings as inputs.

If I change https://github.com/pytorch/pytorch/blob/15d4dedbbff1ebe4e61d4e07b32576b99ee1c9a9/torch/testing/_internal/opinfo/definitions/linalg.py#L936

to use Python stdlib `random.choice` instead of numpy, the test passes and no graph breaks happen.

Here's a minimal reproducer:
```
import torch
import numpy as np
import random

@torch.compile()
def foo(A):
    # Change me to random.choice instead of np.random.choice to work!
    kwargs = {"UPLO": np.random.choice(["L", "U"])}
    return torch.linalg.eigh(A, **kwargs)

foo(
    torch.tensor(np.random.randint(1, 100, size=(8, 8)),
    dtype=torch.float32)
)
```

**Comment by lezcano** [comment]
So this is an issue with the NumPy compat layer. @ev-br this one should be reasonably easy.

**Comment by atalman** [comment]
will be included in 2.2
